### PR TITLE
Helidon config documentation updates

### PR DESCRIPTION
**Pull request by arjav-desai**
Addresses #2850 and #3171

### CODE REVIEW

File: docs/mp/config/01_introduction.adoc [hunk 1]
```diff
@@ -41,13 +41,11 @@ include::{common-deps-page-prefix-inc}[tag=maven-dependency]
         </dependency>
 ----
 
-== About {spec-name}
+== Overview
```
Comment (tomas-langer) [inline docs/mp/config/01_introduction.adoc:44]: Why is this changed? I thought this was used in other docs as well.

File: docs/mp/config/01_introduction.adoc [hunk 2]
```diff
@@ -133,52 +136,12 @@ client_secret=${CLEAR=known_password}
...
-<5> Loads the system properties config source.
-<6> Loads a properties file
-<7> Location of the file: `/conf/prod.properties` on the file system
-<8> Custom ordinal, if not defined, the value defined in the file, or default value is used
-<9> The file is optional (if not optional and no file is found, the bootstrap fails)
-<10> Loads a YAML file
-<11> Location of the file: `META-INF/database.yaml` on the classpath
-
-For backward compatibility, we will support usage of Helidon SE meta-configuration until version 3.0.0. Using this approach causes behavior that is not compatible with {spec-name} specification.
+NOTE: For backward compatibility, we will support usage of Helidon SE meta-configuration until version 3.0.0. Using this approach causes behavior that is not compatible with {spec-name} specification.
```
Comment (tomas-langer) [inline docs/mp/config/01_introduction.adoc:144]: You have removed the MP Meta configuration. That should not have been removed - this is a Helidon MP specific feature (and is not related to SE meta configuration).

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: docs/mp/config/01_introduction.adoc (hunk 2 above)
Comment (arjav-desai) [reply docs/mp/config/01_introduction.adoc:144]: I have just moved it from here to second chapter which is more of example and linked in from this section.

File: docs/mp/config/01_introduction.adoc (hunk 1 above)
Comment (arjav-desai) [reply docs/mp/config/01_introduction.adoc]: It different across chapters e.g. sometimes just spec-name, sometimes overview, sometimes introductions etc. For this, I reverted to old way.

**Review by tomas-langer — APPROVED** [review]
LGTM
